File: mybatis-plus-join-test/test-springboot3-jdk17/src/main/java/com/yulichang/test/springboot3jdk17/entity/UserDO.java
package com.leheyue.test.springboot3jdk17.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import com.leheyue.test.springboot3jdk17.enums.Sex;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Map;

@Data
@ToString
@Accessors(chain = true)
@EqualsAndHashCode
@TableName(value = "`user`", autoResultMap = true)
public class UserDO implements Serializable {

    @TableId
    private Integer id;

    private Integer pid;

    @TableField("`name`")
    private String name;

    @TableField(value = "`json`", typeHandler = JacksonTypeHandler.class)
    private Map<String, String> json;

    private Sex sex;

    @TableField("head_img")
    private String img;

    private LocalDateTime createTime;

    private Integer addressId;

    private Integer addressId2;

    @TableLogic
    private Boolean del;

    private Integer createBy;

    @TableField(exist = false)
    private String createName;

    private Integer updateBy;

    @TableField(exist = false)
    private String updateName;


    @TableField(exist = false)
    private String alias;

    @TableField(exist = false)
    private List<UserDO> children;

    @TableField(exist = false)
    private List<AddressDO> addressList;

    @TableField(exist = false)
    private List<AddressDO> addressList2;
}

File: mybatis-plus-join-test/test-kotlin/src/main/kotlin/com/github/yulichang/test/kt/entity/UserDO.kt
package com.github.leheyue.test.kt.entity

import com.baomidou.mybatisplus.annotation.TableField
import com.baomidou.mybatisplus.annotation.TableLogic
import com.baomidou.mybatisplus.annotation.TableName
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler
import com.github.leheyue.annotation.DynamicTableName
import com.github.leheyue.test.kt.enums.Sex
import lombok.experimental.FieldNameConstants
import java.io.Serializable
import java.time.LocalDateTime

@Suppress("unused")
@DynamicTableName
@FieldNameConstants
@TableName(value = "`user`", autoResultMap = true)
open class UserDO : ID<Int?>(), Serializable {
    var pid: Int? = null

    @TableField("`name`")
    var name: String? = null

    @TableField(value = "`json`", typeHandler = JacksonTypeHandler::class)
    var json: Map<String, String>? = null
    var sex: Sex? = null

    @TableField("head_img")
    var img: String? = null
    var createTime: LocalDateTime? = null
    var addressId: Int? = null
    var addressId2: Int? = null

    @TableLogic
    var del: Boolean? = null
    var createBy: Int? = null

    @TableField(exist = false)
    var createName: String? = null
    var updateBy: Int? = null

    @TableField(exist = false)
    var updateName: String? = null

    @TableField(exist = false)
    var alias: String? = null

    @TableField(exist = false)
    var children: List<UserDO>? = null

    @TableField(exist = false)
    var addressList: List<AddressDO>? = null

    @TableField(exist = false)
    var addressList2: List<AddressDO>? = null
}
